src/main/java/life/qbic/App.java

```diff
@@ -37,12 +38,34 @@ public static void main(String[] args) throws IOException {
    */
   private static QbicDataDownloader loginToOpenBIS(
       PostmanCommandLineOptions commandLineParameters) {
-    LOG.info(String.format("Please provide password for user '%s':", commandLineParameters.user));
 
-    String password = OpenBISPasswordParser.readPasswordFromConsole();
+    String password = "";
+
+    //Password read in from environment Variable
+    if ((!(commandLineParameters.environmentVariableName.isEmpty()))) {
+
+      if (System.getenv(commandLineParameters.environmentVariableName) == null) {
+
+        LOG.info(String.format("Unfortunately, the given environment variable does not exist. " +
+                "Please provide password for user '%s':", commandLineParameters.user));
+
+        password = OpenBISPasswordParser.readPasswordFromConsole();
+      }
+      else {
+        password = System.getenv(commandLineParameters.environmentVariableName);
+      }
+    }
```

Comment:
Another way to write this if/else block would be 
```
      String password = System.getenv(commandLineParameters.environmentVariableName) // might be null
      if (password == null) {
          LOG.info(String.format("Unfortunately, the given environment variable does not exist. " +
                "Please provide password for user '%s':", commandLineParameters.user));
          OpenBISPasswordParser.readPasswordFromConsole()
      }
    }
```
I like your code. Just wanted to point this out.
Javas Optional could come in handy here as well. Although the Optional approach reduces the lines of code, you would lose verbosity to the user.
```
      Optional<String> envPassword = Optional.ofNullable("Test");
      password = envPassword.orElseGet(OpenBISPasswordParser::readPasswordFromConsole);
    }
```